RavenDB-13840 attempted to delete value but it wasn't in index

karmeli87 opened the pull request:
seems to fix the repro.
this is draft to let the test suit run.

karmeli87 committed:
RavenDB-13840 [WIP] attempted to delete value but it wasn't in index

seems to fix the repro.
this is draft to let the test suit run.

karmeli87 commented:
I think that regardless to this fix we have AVE in 32 bit :/

ayende commented on src/Raven.Server/Config/Categories/StorageConfiguration.cs:
```diff
@@ -13,7 +13,7 @@ public class StorageConfiguration : ConfigurationCategory
 
         [Description("Use the 32 bits memory mapped pager, even when running in 64 bits")]
-        [DefaultValue(false)]
+        [DefaultValue(true)]
```
Need to revert this

karmeli87 replied:
This is draft and I want it to run on 32 bits (64 is by default)

ayende reviewed (commented):
The issue is that we are deleting the active section and then allocating from it?

karmeli87 commented:
I'm still not 100% sure, but... 
The way we are getting the `ActiveDataSmallSection` is as following:
https://github.com/ravendb/ravendb/blob/71ea31731fbd3ead4aa0d09ef2902d4ba0d5340c/src/Voron/Data/Tables/Table.cs#L44

And when we delete a section we forget to update this value, so in the next tx we loading the wrong page.

karmeli87 committed:
RavenDB-13840 [WIP] attempted to delete value but it wasn't in index

This is what presumably happened:

in the same tx
1. we tried to allocate from the `ActiveDataSmallSection` and failed
2. we replaced it with some candidate (and don't update his page in the tree!)
3. we freed the section that failed to allocate
4. we end the tx
5. we start a new tx and load the wrong `ActiveDataSmallSection`

karmeli87 committed:
RavenDB-13840 add failing test

karmeli87 commented:
Closing and reopen in 4.1 since is has to be done there also. (https://github.com/ravendb/ravendb/pull/9520)
